### PR TITLE
libsimdpp 2.1 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
libsimdpp 2.1 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4037]
- dev_url: https://github.com/p12tic/libsimdpp/tree/v2.1

### Explanation of changes:

- headers only package
- add tests for each header win/unix in `run_test.[sh,bat]`

[PKG-4037]: https://anaconda.atlassian.net/browse/PKG-4037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -0,0 +1,14 @@
+:: Generate the build files.
+cmake . -B build -G"Ninja" %CMAKE_ARGS%      ^
+    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX%  ^
+    -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX%     ^
+    -DCMAKE_BUILD_TYPE=Release
+if errorlevel 1 exit 1
+
+:: Build.
+cmake --build build -j %CPU_COUNT% --config Release
```
Comment (JeanChristopheMorinPerso) [inline recipe/bld.bat]:
```suggestion
cmake --build build -j %CPU_COUNT%
```
`--config` is for multi-configuration generators ("Visual Studio XX XXX", "Ninja Multi-Config"). See https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-build-config.

Multi-configuration generators allow to generate the build system once and get multiple configurations (Release, Debug, etc) all in one go. So in a multi-config scenario, you basically do `cmake -G<generator> .` and then `cmake --build . --config Release` and you could also use `--config Debug` for example.

File: recipe/run_test.bat [hunk 2]
```diff
@@ -0,0 +1,384 @@
+
+:: The environment variables 'LIBSIMDPP_NAME_VERSION' is
+:: being passed through. If you are splitting build and test phases, 
+:: ensure that this value is also set similarly at test time.
+echo on
+
+if not exist %PREFIX%\Library\lib\pkgconfig\%LIBSIMDPP_NAME_VERSION%.pc exit 1
+if not exist %PREFIX%\Library\include\libsimdpp-2.1\simdpp\capabilities.h exit 1
```
Comment (JeanChristopheMorinPerso) [inline recipe/run_test.bat]: I think most of these are a little bit overkill. It's fine to check for the pc file and a couple of headers, but there is no value in checking that all headers exist.

Comment (lorepirri) [reply recipe/run_test.bat]: ok, makes sense, I tool the first, middle, and last one and the `.pc` file

File: recipe/bld.bat (hunk 1 above)
Comment (lorepirri) [reply recipe/bld.bat]: I removed it also from the `install`, thank you!

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by psteyer — APPROVED** [review]

**Pull request merged**